applications/ShallowWaterApplication/custom_utilities/replicate_model_part_utility.cpp

```diff
@@ -0,0 +1,238 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "replicate_model_part_utility.h"
+#include "utilities/assign_unique_model_part_collection_tag_utility.h"
+
+
+namespace Kratos
+{
+
+ReplicateModelPartUtility::ReplicateModelPartUtility(
+    ModelPart& rOriginModelPart,
+    ModelPart& rDestinationModelPart,
+    bool ReplicateSubModelParts
+) : mrOriginModelPart(rOriginModelPart)
+  , mrDestinationModelPart(rDestinationModelPart)
+{
```

Comment:
```suggestion
, mReplicateSubModelParts(ReplicateSubModelParts)
{
```
To make the member variable `const bool`

Reply:
done